tutorials/dataframe/df027_SQliteDependencyOverVersion.C

```diff
@@ -13,47 +13,50 @@
 /// \macro_image
 ///
 /// \date August 2018
-/// \author Alexandra-Maria Dobrescu
+/// \authors Alexandra-Maria Dobrescu, Sergey Linev
 
-void df027_SQliteDependencyOverVersion () {
 
+void df027_SQliteDependencyOverVersion ()
+{
    auto rdfb = ROOT::RDF::FromSqlite("http://root.cern/files/root_download_stats.sqlite", "SELECT * FROM accesslog;");
 
    auto minTimeStr = *rdfb.Reduce([](std::string a, std::string b) {return std::min(a, b);}, "Time", std::string("Z"));
 
    std::cout << "Minimum time is '" << minTimeStr << "'" << std::endl;
 
    double minTime = TDatime(minTimeStr.c_str()).Convert();
-   double now = TDatime().Convert();
+   double maxTime = minTime + 3600.*24*365.25*4; // cover approx 4 years from minimal time
 
    auto rdf = rdfb.Define("datime", [](const std::string &time){return TDatime(time.c_str()).Convert();}, {"Time"});
 
    auto h614 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.14");}, {"Version"})
-                  .Histo1D({"h614", "Download time for version 6.14", 16, minTime, now}, {"datime"});
+                  .Histo1D({"h614", "Download time for version 6.14", 64, minTime, maxTime}, {"datime"});
 
    auto h616 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.16");}, {"Version"})
-                  .Histo1D({"h616", "Download time for version 6.16", 16, minTime, now}, {"datime"});
+                  .Histo1D({"h616", "Download time for version 6.16", 64, minTime, maxTime}, {"datime"});
 
    auto h618 = rdf.Filter([](const std::string &v){ return 0 == v.find("6.18");}, {"Version"})
-                  .Histo1D({"h618", "Download time for version 6.18", 16, minTime, now}, {"datime"});
-
-   // Add here a newer version!
+                  .Histo1D({"h618", "Download time for version 6.18", 64, minTime, maxTime}, {"datime"});
 
    auto histoList = {h614, h616, h618};
-   auto canvases = new std::vector<TCanvas*>(histoList.size());
 
-   gStyle->SetTimeOffset(0);
-   gStyle->SetOptStat(0);
-   auto histoIdx = 0U;
+   std::cout << "Be aware that rdf.Filter() execution takes ~30s" << std::endl;
+
+   std::vector<TObject*> drawHistos;
+
    for (auto histo : histoList) {
```

Comment:
```suggestion
   for (auto &&histo : histoList) {
```

Reply:
This produce a lot of compiler errors:
```
tutorials/dataframe/df027_SQliteDependencyOverVersion.C:46:12: error: no viable overloaded 'operator->'
      histo->GetXaxis()->SetTimeDisplay(1);
      ~~~~~^
include/ROOT/RResultPtr.hxx:249:7: note: candidate function not viable: 'this' argument has type 'const ROOT::RDF::RResultPtr<TH1D>', but method is not marked const
   T *operator->()
      ^
In file included from input_line_8:1:
```

Reply:
Yes because the body of the for loop is actually modifying the histograms in place, then just
```
for (auto &histo : histoList)
```

Reply:
Getting same error

Reply:
I guess, list of RDF histograms is const by default.

Reply:
> I guess, list of RDF histograms is const by default.

False. This tutorial is abusing imho the `std::initializer_list` for only 3 variables. I'd rather the calls to the 3 histograms separately, can be easily factorizable in a function that takes an histogram input by reference. Or if you like more the approach iterating over the histograms then create a vector of `RResultPtr` or `TH1D` directly (it's an extra copy, but it's anyway necessary for the proper functioning of the graphics so doesn't harm more than it already is doing)

Reply:
This was original code which I did not touch. 
Should I modify it or you will do it later?